providers/posflag/posflag.go

```diff
@@ -43,11 +53,25 @@ func Provider(f *pflag.FlagSet, delim string, ko *koanf.Koanf) *Posflag {
 // This is useful for cases where complex types like slices separated by
 // custom separators.
 func ProviderWithValue(f *pflag.FlagSet, delim string, ko *koanf.Koanf, cb func(key string, value string) (string, interface{})) *Posflag {
```

Comment:
If the functional option refactor of Provider() is accepted, then perhaps ProviderWithValue can be removed/deprecated depending on the importance of backward compatibility.